src/client/dfs/dfs.c

```diff
@@ -3447,8 +3447,8 @@ dfs_lookup(dfs_t *dfs, const char *path, int flags, dfs_obj_t **_obj,
 }
 
 int
-dfs_readdir(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor, uint32_t *nr,
-	    struct dirent *dirs)
+readdir_int(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor, uint32_t *nr,
```

Comment:
You could also just call this dfs_readdirplus() and have dfs_readdir() call it with a NULL value for stbufs to avoid a second wrapper function later on.

Reply:
although we don't adhere to this much, but we shouldn't call API functions internally, in case we add some API tracing later.